core/arch/arm/plat-stm32mp1/drivers/stm32mp1_pmic.c

```diff
@@ -436,6 +438,79 @@ static void save_power_configurations(void)
 			     regu_lp_state[n].name);
 }
 
+/* Preallocate not that much regu references */
+static char *nsec_access_regu_name[PMIC_REGU_COUNT];
+
+bool stm32mp_nsec_can_access_pmic_regu(const char *name)
+{
+	size_t n = 0;
+
+	for (n = 0; n < ARRAY_SIZE(nsec_access_regu_name); n++)
+		if (nsec_access_regu_name[n] &&
+		    !strcmp(nsec_access_regu_name[n], name))
+			return true;
+
+	return false;
+}
+
+static void register_nsec_regu(const char *name_ref)
+{
+	size_t n = 0;
+
+	assert(!stm32mp_nsec_can_access_pmic_regu(name_ref));
+
+	for (n = 0; n < ARRAY_SIZE(nsec_access_regu_name); n++) {
+		if (nsec_access_regu_name[n])
+			continue;
+
+		nsec_access_regu_name[n] = calloc(1, strlen(name_ref) + 1);
+		assert(nsec_access_regu_name[n]);
+		strcpy(nsec_access_regu_name[n], name_ref);
+		break;
+	}
```

Comment:
What if there is no more room? I suggest `assert(n < ARRAY_SIZE(nsec_access_regu_name));` here.

Reply:
The assert below check the regu name is registered (thus alloced).

Reply:
That's right, ignore my comment.